tests/Unit/Evolution/DgSubcell/Test_Projection.cpp

```diff
@@ -0,0 +1,154 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <algorithm>
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Evolution/DgSubcell/Projection.hpp"
+#include "Helpers/Evolution/DgSubcell/ProjectionTestHelpers.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+namespace Tags {
+template <typename Tag>
+struct Prefix : db::PrefixTag {
+  using tag = Tag;
+  using type = typename Tag::type;
+};
+
+struct Scalar : db::SimpleTag {
+  using type = ::Scalar<DataVector>;
+};
+
+template <size_t Dim>
+struct Vector : db::SimpleTag {
+  using type = tnsr::I<DataVector, Dim>;
+};
+}  // namespace Tags
+
+template <size_t MaxPts, size_t Dim, Spectral::Basis BasisType,
+          Spectral::Quadrature QuadratureType>
+void test_project_fd() noexcept {
+  CAPTURE(Dim);
+  CAPTURE(BasisType);
+  CAPTURE(QuadratureType);
+  static_assert(Spectral::maximum_number_of_points<BasisType> >= MaxPts,
+                "Basis does not have enough points. Must have at least 8 "
+                "points per element");
+
+  for (size_t num_pts_1d = std::max(
+           static_cast<size_t>(2),
+           Spectral::minimum_number_of_points<BasisType, QuadratureType>);
+       num_pts_1d < MaxPts + 1; ++num_pts_1d) {
+    CAPTURE(num_pts_1d);
+    const Mesh<Dim> dg_mesh{num_pts_1d, BasisType, QuadratureType};
+    const auto logical_coords = logical_coordinates(dg_mesh);
+    const size_t num_subcells_1d = 2 * num_pts_1d - 1;
+    const Mesh<Dim> subcell_mesh(num_subcells_1d,
+                                 Spectral::Basis::FiniteDifference,
+                                 Spectral::Quadrature::CellCentered);
+    const DataVector nodal_coeffs =
+        TestHelpers::evolution::dg::subcell::cell_values(dg_mesh.extents(0) - 2,
+                                                         logical_coords);
+    // Test projection of a DataVector
+    const DataVector cell_average_times_volume =
+        evolution::dg::subcell::fd::project(nodal_coeffs, dg_mesh,
+                                            subcell_mesh.extents());
+    CHECK_ITERABLE_APPROX(
+        cell_average_times_volume,
+        TestHelpers::evolution::dg::subcell::cell_values(
+            dg_mesh.extents(0) - 2, logical_coordinates(subcell_mesh)));
```

Comment:
I suggest assigning this to a variable (perhaps `expected_subcell_coeffs` or whatever you like), so you can reuse here + twice below in the Vars helper lambda